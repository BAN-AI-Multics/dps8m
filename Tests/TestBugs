de DSBR.ADDR 01000
de dsbr.u 1
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM

load Play/TestBugs.oct segment 0 address 0
load Play/TestBugsA.oct deferred

load Play/UtilsA.oct deferred
load Play/simh_operators.oct deferred

segments lot create
segments create stack 0
segments resolve
segments load deferred
segments lot snap

;set debug stderr
;set cpu debug=trace;regdumpaqi;regdumpidx;addrmod;appending;REGDUMPPR
;set cpu debug=trace;addrmod;appending;REGDUMPPR

g 
q

