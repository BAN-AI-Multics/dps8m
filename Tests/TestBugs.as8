
    option  Multics     " Multics style calling conventions

    entrypoint main
    
	name	main
	segdef	main

	segref	UtilsA, ioa,rd_clockA
	segref	lot_,lot_
    segref  stack_0, stack_0

    segref  Bug_B_Gone_1, _main_BBG_1      " 1st bug regression module ...


    link    I, <UtilsA>|[ioa]
    link    D, <UtilsA>|[rd_clockA]

main:

    epp2	l43170,n*
    puls1


    tsx2    setupFXE
    tra     call_main
    "push

    LDA     I
    tsp2	pr4|I, *        " ioa
    acc     /This is "Bug-B-Gone" (die bugs, die, die) - a.k.a 'TestBugs'! - It's /

    tsp2    rd_clockA
    "tsp2    <UtilsA>|[rd_clockA]
    "tsp2    lp|D,*
    "tsp2    UtilsA$rd_clockA

    staq	Date        " MM/DD HH:MM
        
    tsp2	ioa
    acc     /'^g^g'\r\n/
    nop     Date
    nop     Date+1

"
"
call_main:

    call    _main_BBG_1(ap|0)

allDone:
    tsp2	ioa
    acc     /T-H-A-T-S all folks (from 0|main) !!! /

	dis
    
"
" data ....
"
even

Date:   bss     2


" If not fault, dis@1
	dis	*

    even

l43170:	
        ITS 9, 16

        oct	000400000043
        oct	000000000000
        


    INCLUDE     setupFXE.as8

	end



