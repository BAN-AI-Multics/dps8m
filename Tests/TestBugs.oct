!INFO date Tue Feb 18 04:16:48 2014 UTC
!INFO src TestBugs.as8
!SIZE 000200
!SEGNAME main
!SEGDEF main 000000
!LINKAGE 000150 12
!SEGREF UtilsA ioa 000150
!SEGREF UtilsA rd_clockA 000152
!SEGREF lot_ lot_ 000154
!SEGREF stack_0 stack_0 000156
!SEGREF Bug_B_Gone_1 _main_BBG_1 000160
!SEGREF Bug_B_Gone_2 _main_BBG_2 000162
!SEGREF simh_operators_ operator_table 000164
!SEGREF simh_operators_ operator_table 000166 +632
!SEGREF simh_operators_ operator_table 000170 +633
!SEGREF simh_operators_ operator_table 000172 +634
!SEGREF simh_operators_ operator_table 000174 +635
!SEGREF simh_operators_ operator_table 000176 +636
!GO 000000
000000 xxxx 000074352020     epp2	l43170,n*
000001 xxxx 000100702000     tsx2    setupFXE
000002 xxxx 000000235000     LDA     I
000003 xxxx 400000272120     tsp2	pr4|I, *        " ioa
000004 xxxx 105124150151     acc     /This is "Bug-B-Gone" (die bugs, die, die) - a.k.a 'TestBugs'! - It's /
000005 xxxx 163040151163 
000006 xxxx 040042102165 
000007 xxxx 147055102055 
000010 xxxx 107157156145 
000011 xxxx 042040050144 
000012 xxxx 151145040142 
000013 xxxx 165147163054 
000014 xxxx 040144151145 
000015 xxxx 054040144151 
000016 xxxx 145051040055 
000017 xxxx 040141056153 
000020 xxxx 056141040047 
000021 xxxx 124145163164 
000022 xxxx 102165147163 
000023 xxxx 047041040055 
000024 xxxx 040111164047 
000025 xxxx 163040040040 
000026 xxxx 400002272120     tsp2    rd_clockA
000027 xxxx 000070757000     staq	Date        " MM/DD HH:MM
000030 xxxx 400000272120     tsp2	ioa
000031 xxxx 010047136147     acc     /'^g^g'\r\n/
000032 xxxx 136147047015 
000033 xxxx 012040040040 
000034 xxxx 000070011000     nop     Date
000035 xxxx 000071011000     nop     Date+1
000036 xxxx 600000254100     call    _main_BBG_1(ap|0)
000037 xxxx 600040753100 
000040 xxxx 000000350100 
000041 xxxx 400010352120 
000042 xxxx 700036670120 
000043 xxxx 600000173100 
000044 xxxx 600040073100 
000045 xxxx 600000254100     call    _main_BBG_2(ap|0)
000046 xxxx 600040753100 
000047 xxxx 000000350100 
000050 xxxx 400012352120 
000051 xxxx 700036670120 
000052 xxxx 600000173100 
000053 xxxx 600040073100 
000054 xxxx 400000272120     tsp2	ioa
000055 xxxx 046124055110     acc     /T-H-A-T-S all folks (from 0|main) !!! /
000056 xxxx 055101055124 
000057 xxxx 055123040141 
000060 xxxx 154154040146 
000061 xxxx 157154153163 
000062 xxxx 040050146162 
000063 xxxx 157155040060 
000064 xxxx 174155141151 
000065 xxxx 156051040041 
000066 xxxx 041041040040 
000067 xxxx 000000616000 	dis
000070 xxxx 000000000000 Date:   bss     2
000072 xxxx 000072616000 	dis	*
000073 xxxx 000000011000     even "(allocating 1 nop)
000074 xxxx 000011000043         ITS 9, 16
000075 xxxx 000020000000 
000076 xxxx 000400000043         oct	000400000043
000077 xxxx 000000000000         oct	000000000000
000100 xxxx 400004764120         lprplp  lot_            " set-up lp to point to this segments linkage section
000101 xxxx 400006372120         eppsp   stack_0         " set-up stack pointer
000102 xxxx 400006373520         eppsb   stack_0         " set-up stack base
000103 xxxx 400006373520     eppsb   stack_0
000104 xxxx 400006372120     eppsp   stack_0
000105 xxxx 000100152003     adwpsp  =0100,du
000106 xxxx 700022652100     sprisp	sb|stack_header.stack_begin_ptr 		" save beginning of 1st stack frame
000107 xxxx 700024652100     sprisp	sb|stack_header.stack_end_ptr
000110 xxxx 400004352120 	eppbp	lot_						" set the lot ptrs
000111 xxxx 700026252100 	spribp	sb|stack_header.lot_ptr
000112 xxxx 700052252100 	spribp	sb|stack_header.isot_ptr
000113 xxxx 000144352020 	eppbp	=its(-2,2),*		" signal ptr terminates process
000114 xxxx 700030252100 	spribp	sb|stack_header.signal_ptr
000115 xxxx 400014352120 	eppbp	simh_operators_$operator_table  		" Get a ptr to  PL/I operators table.
000116 xxxx 700034252100 	spribp	sb|stack_header.pl1_operators_ptr 	" Store in  inzr_stk0
000117 xxxx 400016352120 	eppbp	simh_operators_$operator_table+call_offset,*
000120 xxxx 700036252100 	spribp	sb|stack_header.call_op_ptr
000121 xxxx 400020352120 	eppbp	simh_operators_$operator_table+push_offset,*
000122 xxxx 700040252100 	spribp	sb|stack_header.push_op_ptr
000123 xxxx 400022352120 	eppbp	simh_operators_$operator_table+return_offset,*
000124 xxxx 700042252100 	spribp	sb|stack_header.return_op_ptr
000125 xxxx 400024352120 	eppbp	simh_operators_$operator_table+return_no_pop_offset,*
000126 xxxx 700044252100 	spribp	sb|stack_header.ret_no_pop_op_ptr
000127 xxxx 400026352120 	eppbp	simh_operators_$operator_table+entry_offset,*
000130 xxxx 700046252100 	spribp	sb|stack_header.entry_op_ptr
000131 xxxx 000146352020 	eppbp	=its(-1,1),*
000132 xxxx 700070252100 	spribp	sb|stack_header.heap_header_ptr
000133 xxxx 700046272120     tsp2 pr7|stack_header.entry_op_ptr,*
000134 xxxx 000060627000     push
000135 xxxx 700040272120 
000136 xxxx 000000710012 	tra		0,2				" ... and return to caller
000137 xxxx 000000011000 	even "(allocating 1 nop)
000140 xxxx 000000000004 no_args:	zero	0,4
000141 xxxx 000000000000 	zero	0,0
000142 xxxx 000000000000 decimal literal (=0)
000143 xxxx 000000000000 decimal literal (=0)
000144 xxxx 077776000043 its literal (=its(77776,000002,00))
000145 xxxx 000002000000 
000146 xxxx 077777000043 its literal (=its(77777,000001,00))
000147 xxxx 000001000000 
000150 xxxx 000000000043 link UtilsA$ioa
000151 xxxx 000000000000 
000152 xxxx 000000000043 link UtilsA$rd_clockA
000153 xxxx 000000000000 
000154 xxxx 000000000043 link lot_$lot_
000155 xxxx 000000000000 
000156 xxxx 000000000043 link stack_0$stack_0
000157 xxxx 000000000000 
000160 xxxx 000000000043 link Bug_B_Gone_1$_main_BBG_1
000161 xxxx 000000000000 
000162 xxxx 000000000043 link Bug_B_Gone_2$_main_BBG_2
000163 xxxx 000000000000 
000164 xxxx 000000000043 link simh_operators_$operator_table
000165 xxxx 000000000000 
000166 xxxx 000000000043 link simh_operators_$operator_table+632
000167 xxxx 001170000000 
000170 xxxx 000000000043 link simh_operators_$operator_table+633
000171 xxxx 001171000000 
000172 xxxx 000000000043 link simh_operators_$operator_table+634
000173 xxxx 001172000000 
000174 xxxx 000000000043 link simh_operators_$operator_table+635
000175 xxxx 001173000000 
000176 xxxx 000000000043 link simh_operators_$operator_table+636
000177 xxxx 001174000000 
