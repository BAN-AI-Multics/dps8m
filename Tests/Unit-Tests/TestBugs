de DSBR.ADDR 01000
de dsbr.u 1
de dsbr.bnd 1
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM

load ./TestBugs.oct segment 0 address 0
load ./TestBugsA.oct deferred
load ./TestBugsB.oct deferred
load ./TestBugs15.oct as segment 015 deferred

load ./UtilsA.oct deferred
load ./simh_operators.oct deferred

segments lot create
segments create stack 0
segments resolve
segments load deferred
segments lot snap

;set debug stderr
;set cpu debug=trace;regdumpaqi;regdumpidx;addrmod;appending;REGDUMPPR
; set cpu debug=trace;addrmod;appending;REGDUMPPR
;set cpu debug=trace

g
q

