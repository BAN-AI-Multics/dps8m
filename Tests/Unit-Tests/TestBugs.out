zero-initialized segments 0 .. 15
zero-initialized SDWAM
segment created for 'main'
segdef created for segment main, symbol 'main', addr:000000
segref created for segment 'UtilsA' symbol:ioa, addr:000150
segref created for segment 'UtilsA' symbol:rd_clockA, addr:000152
segref created for segment 'lot_' symbol:lot_, addr:000154
segref created for segment 'stack_0' symbol:stack_0, addr:000156
segref created for segment 'Bug_B_Gone_1' symbol:_main_BBG_1, addr:000160
segref created for segment 'Bug_B_Gone_2' symbol:_main_BBG_2, addr:000162
segref created for segment 'simh_operators_' symbol:operator_table, addr:000164
segref created for segment 'simh_operators_' symbol:operator_table+632, addr:000166
segref created for segment 'simh_operators_' symbol:operator_table+633, addr:000170
segref created for segment 'simh_operators_' symbol:operator_table+634, addr:000172
segref created for segment 'simh_operators_' symbol:operator_table+635, addr:000174
segref created for segment 'simh_operators_' symbol:operator_table+636, addr:000176
Writing SDW to address 00001000 (DSBR.ADDR+2*0 offset) 
127 (000177) words loaded into segment 0(0) at address 000000
segment created for 'Bug_B_Gone_1'
segdef created for segment Bug_B_Gone_1, symbol 'TRA1', addr:000233
entrypoint created for segment main, symbol '_main_BBG_1', addr:000270
segref created for segment 'UtilsA' symbol:ioa, addr:000302
194 (000302) words loaded into segment Bug_B_Gone_1
segment created for 'Bug_B_Gone_2'
entrypoint created for segment main, symbol '_main_BBG_2', addr:000052
segref created for segment 'UtilsA' symbol:ioa, addr:000056
48 (000060) words loaded into segment Bug_B_Gone_2
segment created for 'Bug_B_Gone_seg15'
2 (000002) words loaded into segment Bug_B_Gone_seg15 as segment 13
segment created for 'UtilsA'
segdef created for segment UtilsA, symbol 'A0', addr:000005
segdef created for segment UtilsA, symbol 'A1', addr:000050
segdef created for segment UtilsA, symbol 'rd_clockA', addr:000207
segdef created for segment UtilsA, symbol 'ioa', addr:000271
857 (001531) words loaded into segment UtilsA
segment created for 'simh_operators_'
segdef created for segment simh_operators_, symbol 'alm_call', addr:001175
segdef created for segment simh_operators_, symbol 'alm_entry', addr:001215
segdef created for segment simh_operators_, symbol 'alm_operators_begin', addr:001175
segdef created for segment simh_operators_, symbol 'alm_operators_end', addr:001237
segdef created for segment simh_operators_, symbol 'alm_push', addr:001201
segdef created for segment simh_operators_, symbol 'alm_return', addr:001225
segdef created for segment simh_operators_, symbol 'alm_return_no_pop', addr:001233
segdef created for segment simh_operators_, symbol 'transfer_vector', addr:000551
segdef created for segment simh_operators_, symbol 'operator_table', addr:000000
40 (000050) words loaded into segment simh_operators_
lot_ segment created with 6 sparse entries.
stack_0 segment created as segment# 16 (20) [DSBR.STACK=0002]
Examining segments ... segment numbering begins at: 17
    Processing segment main...
        Resolving segref UtilsA$ioa...found ioa (000150)
            ITS Pair: [even:000021000043, odd:000271000000]
        Resolving segref UtilsA$rd_clockA...found rd_clockA (000152)
            ITS Pair: [even:000021000043, odd:000207000000]
        Resolving segref lot_$lot_...found lot_ (000154)
            ITS Pair: [even:000022000043, odd:000000000000]
        Resolving segref stack_0$stack_0...found stack_0 (000156)
            ITS Pair: [even:000020000043, odd:000000000000]
        Resolving segref Bug_B_Gone_1$_main_BBG_1...found _main_BBG_1 (000160)
            ITS Pair: [even:000023000043, odd:000270000000]
        Resolving segref Bug_B_Gone_2$_main_BBG_2...found _main_BBG_2 (000162)
            ITS Pair: [even:000024000043, odd:000052000000]
        Resolving segref simh_operators_$operator_table...found operator_table (000164)
            ITS Pair: [even:000025000043, odd:000000000000]
        Resolving segref simh_operators_$operator_table...found operator_table+632 (000166)
            ITS Pair: [even:000025000043, odd:001170000000]
        Resolving segref simh_operators_$operator_table...found operator_table+633 (000170)
            ITS Pair: [even:000025000043, odd:001171000000]
        Resolving segref simh_operators_$operator_table...found operator_table+634 (000172)
            ITS Pair: [even:000025000043, odd:001172000000]
        Resolving segref simh_operators_$operator_table...found operator_table+635 (000174)
            ITS Pair: [even:000025000043, odd:001173000000]
        Resolving segref simh_operators_$operator_table...found operator_table+636 (000176)
            ITS Pair: [even:000025000043, odd:001174000000]
    Processing segment Bug_B_Gone_1...
        Resolving segref UtilsA$ioa...found ioa (000302)
            ITS Pair: [even:000021000043, odd:000271000000]
    Processing segment Bug_B_Gone_2...
        Resolving segref UtilsA$ioa...found ioa (000056)
            ITS Pair: [even:000021000043, odd:000271000000]
    Processing segment Bug_B_Gone_seg15...
    Processing segment UtilsA...
    Processing segment simh_operators_...
    Processing segment lot_...
    Processing segment stack_0...
Loading deferred segments ...
    loading Bug_B_Gone_1 as segment# 023
Writing SDW to address 00001046 (DSBR.ADDR+2*19 offset) 
      196 (000304) words loaded into segment 19 (23) at address 201000
    loading Bug_B_Gone_2 as segment# 024
Writing SDW to address 00001050 (DSBR.ADDR+2*20 offset) 
      48 (000060) words loaded into segment 20 (24) at address 201320
    loading Bug_B_Gone_seg15 as segment# 015
Writing SDW to address 00001032 (DSBR.ADDR+2*13 offset) 
      258 (000402) words loaded into segment 13 (15) at address 201400
    loading UtilsA as segment# 021
Writing SDW to address 00001042 (DSBR.ADDR+2*17 offset) 
      1064 (002050) words loaded into segment 17 (21) at address 202020
    loading simh_operators_ as segment# 025
Writing SDW to address 00001052 (DSBR.ADDR+2*21 offset) 
      671 (001237) words loaded into segment 21 (25) at address 204100
    loading lot_ as segment# 022
Writing SDW to address 00001044 (DSBR.ADDR+2*18 offset) 
      4095 (007777) words loaded into segment 18 (22) at address 205340
LOT => PR[4]: SNR=00022 RNR=0 WORDNO=000000 BITNO:00
LOT => PR[5]: SNR=00022 RNR=0 WORDNO=000000 BITNO:00
    loading stack_0 as segment# 020
Writing SDW to address 00001040 (DSBR.ADDR+2*16 offset) 
      49152 (140000) words loaded into segment 16 (20) at address 215340
snapping lot_ links...

dps-8/m simulator V4.0-0 Beta        git commit id: c8ae76cb
	205340 + 0 => 000000000150
	205340 + 23 => 000023000302
	205340 + 24 => 000024000056
This is "Bug-B-Gone" (die bugs, die, die) - a.k.a 'TestBugs'! - It's ' 6/19 10:06 '
BBG-1 ...
ORSQ test 0 ... 0 (Should be 0)
ORSQ test 1 ... 1234567 (Should be 1234567)
ORSQ test 2 ... 1234567 (Should be 1234567)
ORSQ test 3 ... 1237567 (Should be 1237567)
ORSQ test 4 ... 7777777 (Should be 7777777)
ORSQ bug >squished<
TRA test ... trying 'tra seg|0,x0'...
    TRA test 0 ... Q=1726354... (pass)
    TRA test 1 ... back frpm TestBugsB...
EPP test 1 ...
BBG-2 ...
ITS test 1 ...
    Result is: 500000 764101 (Should be  5 00000 7641 01)
T-H-A-T-S all folks (from 0|main) !!! 
cpuCycles = 12739
TestBugs-31> g

DIS instruction, IC: main|66 (00000066) (000000616000)
Goodbye
